assembly/src/assembler/mod.rs

```diff
@@ -412,7 +445,7 @@ impl Assembler {
             let is_entry = entrypoint == Some(procedure_gid);
 
             // Fetch procedure metadata from the graph
-            let module = &self.module_graph[procedure_gid.module];
+            let module = &self.module_graph[procedure_gid.module].unwrap_ast();
```

Comment:
Similar question here and in a few other places in this file: are we guaranteed to get an AST module?

Reply:
Documented all valid cases, and removed 2 (which includes the one in `NameResolver`)